bus-mapping/src/evm/opcodes.rs

```diff
@@ -73,155 +81,164 @@ pub trait Opcode: Debug {
+        OpcodeId::MSTORE => &Mstore::<false>,
+        OpcodeId::MSTORE8 => &Mstore::<true>,
+        OpcodeId::SLOAD => &Sload,
+        OpcodeId::SSTORE => &Sstore,
+        OpcodeId::JUMP => &StackOnlyOpcode::<1, 0>,
+        OpcodeId::JUMPI => &StackOnlyOpcode::<2, 0>,
+        OpcodeId::PC => &StackOnlyOpcode::<0, 1>,
+        OpcodeId::MSIZE => &StackOnlyOpcode::<0, 1>,
+        OpcodeId::GAS => &StackOnlyOpcode::<0, 1>,
+        OpcodeId::JUMPDEST => &Dummy,
+        OpcodeId::DUP1 => &Dup::<1>,
+        OpcodeId::DUP2 => &Dup::<2>,
+        OpcodeId::DUP3 => &Dup::<3>,
+        OpcodeId::DUP4 => &Dup::<4>,
+        OpcodeId::DUP5 => &Dup::<5>,
+        OpcodeId::DUP6 => &Dup::<6>,
+        OpcodeId::DUP7 => &Dup::<7>,
+        OpcodeId::DUP8 => &Dup::<8>,
+        OpcodeId::DUP9 => &Dup::<9>,
+        OpcodeId::DUP10 => &Dup::<10>,
+        OpcodeId::DUP11 => &Dup::<11>,
+        OpcodeId::DUP12 => &Dup::<12>,
+        OpcodeId::DUP13 => &Dup::<13>,
+        OpcodeId::DUP14 => &Dup::<14>,
+        OpcodeId::DUP15 => &Dup::<15>,
+        OpcodeId::DUP16 => &Dup::<16>,
+        OpcodeId::SWAP1 => &Swap::<1>,
+        OpcodeId::SWAP2 => &Swap::<2>,
+        OpcodeId::SWAP3 => &Swap::<3>,
+        OpcodeId::SWAP4 => &Swap::<4>,
+        OpcodeId::SWAP5 => &Swap::<5>,
+        OpcodeId::SWAP6 => &Swap::<6>,
+        OpcodeId::SWAP7 => &Swap::<7>,
+        OpcodeId::SWAP8 => &Swap::<8>,
+        OpcodeId::SWAP9 => &Swap::<9>,
+        OpcodeId::SWAP10 => &Swap::<10>,
+        OpcodeId::SWAP11 => &Swap::<11>,
+        OpcodeId::SWAP12 => &Swap::<12>,
+        OpcodeId::SWAP13 => &Swap::<13>,
+        OpcodeId::SWAP14 => &Swap::<14>,
+        OpcodeId::SWAP15 => &Swap::<15>,
+        OpcodeId::SWAP16 => &Swap::<16>,
+        OpcodeId::LOG0 => &Log,
+        OpcodeId::LOG1 => &Log,
+        OpcodeId::LOG2 => &Log,
+        OpcodeId::LOG3 => &Log,
+        OpcodeId::LOG4 => &Log,
         // OpcodeId::CREATE => {},
-        OpcodeId::CALL => Call::gen_associated_ops,
+        OpcodeId::CALL => &Call,
         // OpcodeId::CALLCODE => {},
-        // TODO: Handle RETURN by its own gen_associated_ops.
-        OpcodeId::RETURN => Stop::gen_associated_ops,
+        OpcodeId::RETURN => &Return,
         // OpcodeId::DELEGATECALL => {},
         // OpcodeId::CREATE2 => {},
         // OpcodeId::STATICCALL => {},
-        // TODO: Handle REVERT by its own gen_associated_ops.
-        OpcodeId::REVERT => Stop::gen_associated_ops,
+        // REVERT is almost the same as RETURN
+        OpcodeId::REVERT => &Return,
         OpcodeId::SELFDESTRUCT => {
             warn!("Using dummy gen_selfdestruct_ops for opcode SELFDESTRUCT");
-            dummy_gen_selfdestruct_ops
+            &DummySelfDestruct
         }
         OpcodeId::CALLCODE | OpcodeId::DELEGATECALL | OpcodeId::STATICCALL => {
             warn!("Using dummy gen_call_ops for opcode {:?}", opcode_id);
-            dummy_gen_call_ops
+            &DummySelfDestruct
```

Comment:
why `DummySelfDestruct` for these call ops?

Reply:
it's a typo